src/main/java/com/minecolonies/coremod/util/BarbarianUtils.java

```diff
@@ -0,0 +1,193 @@
+                break;
+            case 3:
+                z -= Configurations.workingRangeTownHall + 20;
+                break;
+            case 4:
+                x += Configurations.workingRangeTownHall + 20;
+                z += Configurations.workingRangeTownHall + 20;
+                break;
+            case 5:
+                x += Configurations.workingRangeTownHall + 20;
+                z -= Configurations.workingRangeTownHall + 20;
+                break;
+            case 6:
+                x -= Configurations.workingRangeTownHall + 20;
+                z += Configurations.workingRangeTownHall + 20;
+                break;
+            case 7:
+                x -= Configurations.workingRangeTownHall + 20;
+                z -= Configurations.workingRangeTownHall + 20;
+                break;
+            default:
+                x += Configurations.workingRangeTownHall + 20;
+                break;
+        }
+
+        //Make sure mob spawns on surface.
+        y = raidingWorld.getTopSolidOrLiquidBlock(new BlockPos.MutableBlockPos(x, y, z)).getY();
+
+        spawn(BarbarianUtils.barbarian, numberOfBarbarians, x, y, z, raidingWorld);
+        spawn(BarbarianUtils.archer, numberOfArcherBarbarians, x, y, z, raidingWorld);
+        spawn(BarbarianUtils.chief, numberOfChiefBarbarians, x, y, z, raidingWorld);
+    }
+
+    /**
+     * Spawns EntityToSPawn at the X, Y, Z
+     */
+
+    public static void spawn(final ResourceLocation entityToSpawn,final int numberOfSpawns,final int x,final int y,final int z, final World world)
+    {
+        IntStream.range(0, numberOfSpawns).forEach($ ->
+        {
+            if (entityToSpawn != null && world != null)
+            {
+                final Entity entity = EntityList.createEntityByIDFromName(entityToSpawn, world);
+                if (entity != null)
+                {
+                    if (entityToSpawn.equals(barbarian))
+                    {
+                        entity.setItemStackToSlot(EntityEquipmentSlot.MAINHAND, new ItemStack(Items.STONE_AXE));
+                    }
+                    if (entityToSpawn.equals(archer))
+                    {
+                        entity.setItemStackToSlot(EntityEquipmentSlot.MAINHAND, new ItemStack(Items.BOW));
+                    }
+                    if (entityToSpawn.equals(chief))
+                    {
+                        entity.setItemStackToSlot(EntityEquipmentSlot.MAINHAND, new ItemStack(Items.GOLDEN_SWORD));
+                        entity.setItemStackToSlot(EntityEquipmentSlot.CHEST, new ItemStack(Items.CHAINMAIL_CHESTPLATE));
+                        entity.setItemStackToSlot(EntityEquipmentSlot.LEGS, new ItemStack(Items.CHAINMAIL_LEGGINGS));
+                        entity.setItemStackToSlot(EntityEquipmentSlot.FEET, new ItemStack(Items.CHAINMAIL_BOOTS));
+                    }
+                    entity.setLocationAndAngles(x, y, z, MathHelper.wrapDegrees(world.rand.nextFloat() * 360.0F), 0.0F);
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Assign this magic number 360.0F to a well-named constant, and use the constant instead. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)